common/reviews/api/rush-lib.api.md

```diff
@@ -127,6 +127,9 @@ export class CredentialCache {
     static usingAsync(options: ICredentialCacheOptions, doActionAsync: (credentialCache: CredentialCache) => Promise<void> | void): Promise<void>;
 }
 
+// @beta (undocumented)
+export function dependencyAnalysis(filename: string, terminal: ITerminal): void;
```

Comment:
Our coding convention is to group APIs into meaningful containers such as a class.  We try to avoid exporting loose functions. (If `dependencyAnalysis()` is a top-level item in a large .api.md file, what "dependency" does it mean? What feature is it related to? It is not obvious.)  Also, the function name should generally start with a verb.

Consider an API design like this:

```ts
/**
 * Exposes the underlying operations of the "rush analyze-telemetry" command.
 */
export class TelemetryAnalysis {
  public static printProjectBuildTimeReport(telemetryFilePath: string, terminal: ITerminal): void;
  public static printCpuCoresReport(telemetryFilePath: string, terminal: ITerminal): void;
}
```

I am also curious how you are using this API.  The caller could simply invoke the `rush analyze-telemetry` shell command -- what is the advantage of calling the TypeScript API?

Reply:
The API design you recommend look great. I will work on this refactoring. Per your last point - are you recommending that the telemetry analysis functions in DependencyAnalysisPlugin.ts could live in AnalyzeAction.ts?

Reply:
This is relevant specifically to the public API surface more than internal usage. If the API isn't exported from the index file (which it may not need to be), use whatever shape is most ergonomic for unit testing and execution.